### PR TITLE
cache GDAL_ONE_BIG_READ value for RasterIO

### PR DESCRIPTION
**Pull request by chacha21**
Proposal for #6726

### CODE REVIEW

File: gcore/gdaldataset.cpp [hunk 1]
```diff
@@ -1968,6 +1969,10 @@ CPLErr GDALDataset::IRasterIO( GDALRWFlag eRWFlag,
 
     CPLAssert(nullptr != pData);
 
+    RawDataset* rawDataset = dynamic_cast<RawDataset*>(this);
```
Comment (rouault) [inline gcore/gdaldataset.cpp:1972]: not pretty to put that here. That should be put in RawDataset::IRasterIO() itself

File: gcore/rawdataset.h [hunk 2]
```diff
@@ -61,9 +61,11 @@ class CPL_DLL RawDataset : public GDALPamDataset
          virtual ~RawDataset() = 0;
 
     bool GetRawBinaryLayout(GDALDataset::RawBinaryLayout&) override;
-
+    void resetCache(void) {this->cachedCPLOneBigReadOption = std::make_pair(false, 0);}
```
Comment (rouault) [inline gcore/rawdataset.h:64]: I feel a better name than resetCache() should be found, as there is a risk of confusion with the block cache.
ClearCachedConfigOption() maybe ?

File: gcore/gdaldataset.cpp (hunk 1 above)
Comment (chacha21) [reply gcore/gdaldataset.cpp:1972]: Ok, inheritance is more complex that I thought. I believed that RawDataset::IRasterIO was somwhat totally overriden

File: gcore/rawdataset.h [hunk 3]
```diff
@@ -61,9 +61,11 @@ class CPL_DLL RawDataset : public GDALPamDataset
          virtual ~RawDataset() = 0;
 
     bool GetRawBinaryLayout(GDALDataset::RawBinaryLayout&) override;
-
+    void ClearCachedConfigOption(void) {this->cachedCPLOneBigReadOption = std::make_pair(false, 0);}
   private:
     CPL_DISALLOW_COPY_ASSIGN(RawDataset)
+  protected:
+    std::pair<bool, int> cachedCPLOneBigReadOption;
```
Comment (rouault) [inline gcore/rawdataset.h:68]: ``#include <utility>`` should be added in the include section at top of this file

**Pull request merged**